.circleci/config.yml

```diff
@@ -602,6 +610,56 @@ defaults:
         MAKEFLAGS: -j 10
         CPUs: 10
 
+  - base_ubuntu2404: &base_ubuntu2404
+      docker:
+        - image: << pipeline.parameters.ubuntu-2404-docker-image >>
+      environment: &base_ubuntu2404_env
+        TERM: xterm
+        MAKEFLAGS: -j 3
+        CPUs: 3
+
+  - base_ubuntu2404_clang: &base_ubuntu2404_clang
+      docker:
+        - image: << pipeline.parameters.ubuntu-2404-clang-docker-image >>
+      environment: &base_ubuntu2404_clang_env
+        TERM: xterm
+        CC: clang
+        CXX: clang++
+        MAKEFLAGS: -j 3
+        CPUs: 3
+
+  - base_ubuntu2404_clang_large: &base_ubuntu2404_clang_large
+      <<: *base_ubuntu2404_clang
+      resource_class: large
+      environment: &base_ubuntu2404_clang_large_env
+        <<: *base_ubuntu2404_clang_env
+        MAKEFLAGS: -j 5
+        CPUs: 5
+
+  - base_ubuntu2404_small: &base_ubuntu2404_small
+      <<: *base_ubuntu2404
+      resource_class: small
+      environment: &base_ubuntu2404_small_env
+        <<: *base_ubuntu2404_env
+        MAKEFLAGS: -j 2
+        CPUs: 2
+
+  - base_ubuntu2404_large: &base_ubuntu2404_large
+      <<: *base_ubuntu2404
+      resource_class: large
+      environment: &base_ubuntu2404_large_env
+        <<: *base_ubuntu2404_env
+        MAKEFLAGS: -j 5
+        CPUs: 5
+
+  - base_ubuntu2404_xlarge: &base_ubuntu2404_xlarge
+      <<: *base_ubuntu2404
+      resource_class: xlarge
+      environment: &base_ubuntu2404_xlarge_env
+        <<: *base_ubuntu2404_env
+        MAKEFLAGS: -j 10
+        CPUs: 10
```

Comment:
We generally add and/remove variants other than the default one as needed. So no need to add them already if nothing's using them.

Having them does not hurt, but the config is already quite long :)

Reply:
Oh, I guess the assumption is that we'll switch all the 2204 ones to them anyway so they'll all be in use soon?

Reply:
Correct - the first step is to merge this PR. Then @rodiazet will rebase his evmone bump PR and switch a few of the failing soltest jobs to use the 2404 images added here (required for the evmone bump to pass). Once his PR is merged, we can then switch the rest of the 2204 jobs to 2404, and perform any required clean-up (i.e. delete the 2204 images).